core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java

```diff
@@ -61,20 +59,17 @@ protected void setTextProvider(TextProvider textProvider) {
         this.textProvider = textProvider;
     }
     
-    @Inject(value="allowStaticMethodAccess", required=false)
-    protected void setAllowStaticMethodAccess(String allowStaticMethodAccess) {
-        this.allowStaticMethodAccess = BooleanUtils.toBoolean(allowStaticMethodAccess);
-    }
-
     public ValueStack createValueStack() {
-        ValueStack stack = new OgnlValueStack(xworkConverter, compoundRootAccessor, textProvider, allowStaticMethodAccess);
+        String allowStaticMethodAccess = container.getInstance(String.class, XWorkConstants.ALLOW_STATIC_METHOD_ACCESS);
+        ValueStack stack = new OgnlValueStack(xworkConverter, compoundRootAccessor, textProvider, Boolean.valueOf(allowStaticMethodAccess));
```

Comment:
```
final boolean allowStaticMethodAccess = BooleanUtils.toBoolean(container.getInstance(String.class, XWorkConstants.ALLOW_STATIC_METHOD_ACCESS)); 
ValueStack stack = new OgnlValueStack(xworkConverter, compoundRootAccessor, textProvider, allowStaticMethodAccess);
```
Now that I'm referring to the correct branch ( :) ) as per my original comment this might be an alternate form for the 2 new code blocks.  
Also ... I forgot that BooleanUtils was present, so it might be better to use it rather than Boolean.parseBoolean.